test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,226 @@
+	if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), routeName, r); err != nil {
+			t.Logf("couldn't get route %s: %v, retrying...", routeName, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to get route: %v", err)
+	}
+
+	// curl to canary, without the Accept-Encoding header set to gzip
+	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+		header, code, err := getHttp(client, r, false)
+		if err != nil {
+			t.Logf("GET %s failed: %v, retrying...", r.Spec.Host, err)
+			return false, nil
+		}
+		if code != http.StatusOK {
+			t.Logf("GET %s failed: status %v, expected %v, retrying...", r.Spec.Host, code, http.StatusOK)
+			return false, nil // retry on 503 as pod/service may not be ready
+		}
+
+		if header.Get("Content-Encoding") == "gzip" {
+			t.Logf("compression is enabled and shouldn't be for %s", r.Name)
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to get results on GET: %v", err)
+	}
+
+	// curl to canary, WITH the Accept-Encoding header set to gzip
+	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+		header, code, err := getHttp(client, r, true)
+		if err != nil {
+			t.Logf("GET %s failed: %v, retrying...", r.Spec.Host, err)
+			return false, nil
+		}
+		if code != http.StatusOK {
+			t.Logf("GET %s failed: status %v, expected %v, retrying...", r.Spec.Host, code, http.StatusOK)
+			return false, nil // retry on 503 as pod/service may not be ready
+		}
+
+		if header.Get("Content-Encoding") != "gzip" {
+			t.Fatalf("compression is not enabled but should be for %s", r.Name)
+		}
+		t.Logf("compression is properly enabled for %s", r.Name)
+
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to get results on GET: %v", err)
+	}
+}
+
+// Get HTTP Headers.  If the request includes a Header of "Accept-Encoding: gzip" and the route
+// mimeType matches the ingress setting, we should see "Content-Encoding: gzip" in the response.
+func getHttp(client *http.Client, route *routev1.Route, addHeader bool) (http.Header, int, error) {
+	// Create the HTTPS request
+	request, err := http.NewRequest("GET", "https://"+route.Spec.Host, nil)
+	if err != nil {
+		return nil, -1, fmt.Errorf("New request failed: %v", err)
+	}
+
+	// Give the instruction to compress
+	if addHeader {
+		request.Header.Add("Accept-Encoding", "gzip")
+	}
+
+	response, err := client.Do(request)
+	if err != nil {
+		return response.Header, response.StatusCode, fmt.Errorf("GET %s failed: %v", route.Spec.Host, err)
+	}
+	// Close response body
+	defer response.Body.Close()
+
+	return response.Header, response.StatusCode, nil
+}
+
+// Update the spec.httpCompressionPolicy with different MIME types
+func testCompressionPolicy(t *testing.T, icName types.NamespacedName, domain string, compressionPolicy operatorv1.HTTPCompressionPolicy) error {
```

Comment:
Call `t.Helper()` at the start of the function.

Reply:
Thanks